cluster/images/provider-terraform/Dockerfile

```
@@ -1,15 +1,27 @@
-FROM hashicorp/terraform:1.2.3
+FROM alpine:3.16.2
+RUN apk --no-cache add ca-certificates bash
 
 ARG TARGETOS
 ARG TARGETARCH
 
-ADD bin/$TARGETOS\_$TARGETARCH/provider /usr/local/bin/crossplane-terraform-provider
+ENV TERRAFORM_VERSION=1.3.4
+# Prevent terraform from forking in case of crash or panic, useful when debugging
+ENV TF_FORK 0
+
+ADD "bin/${TARGETOS}_${TARGETARCH}/provider" /usr/local/bin/crossplane-terraform-provider
 ADD .gitconfig .gitconfig
 
+ADD https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip /tmp
+
+RUN unzip /tmp/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip -d /usr/local/bin \
+  && chmod +x /usr/local/bin/terraform \
+  && rm /tmp/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip
```

```suggestion
RUN cd /tmp \
  && wget https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip \
  && chmod +x /usr/local/bin/terraform \
  && rm /tmp/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip
```

Motivation: download terraform zip, unpack and remove within a single operation so it will not consume 18 MB of redundant image layer

Before:
```
du -sh ./_output/xpkg/linux_amd64/provider-terraform-v0.5.0-rc.0.12.g53d3e81.xpkg
 58M	_output/xpkg/linux_amd64/provider-terraform-v0.5.0-rc.0.12.g53d3e81.xpkg
```

After:
```
du -sh ./_output/xpkg/linux_amd64/provider-terraform-v0.5.0-rc.0.12.g53d3e81.dirty.xpkg
 40M	./_output/xpkg/linux_amd64/provider-terraform-v0.5.0-rc.0.12.g53d3e81.dirty.xpkg
```